### PR TITLE
add try except around onnxruntime if not properly installed

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR adds a try, except around onnxruntime which can be wrongly installed. 

Fixes # (issue)

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SeanNaren — APPROVED** [review]

File: tests/models/test_onnx.py [hunk 1]
```diff
@@ -14,10 +14,16 @@
 import os
 
 import numpy as np
-import onnxruntime
 import pytest
 import torch
 
+try:
+    import onnxruntime
```
Comment (Borda) [inline tests/models/test_onnx.py:21]: Why we require it for testing... it does not make sense to ignore this testing...

**Review by Borda — CHANGES_REQUESTED** [review]

File: tests/models/test_onnx.py (hunk 1 above)
Comment (SeanNaren) [reply tests/models/test_onnx.py:21]: I rain into install issues on mac if I recall correctly, @tchaton is this what you're trying to address here?

Comment (SeanNaren) [reply tests/models/test_onnx.py:21]: I rain into install issues on mac if I recall correctly, @tchaton is this what you're trying to address here?

Comment (Borda) [reply tests/models/test_onnx.py:21]: what issue, can you share details?

Comment (tchaton) [reply tests/models/test_onnx.py:21]:
```
(.venv) ➜  pytorch-lightning git:(bugfix/dict_wo_loss_error) ✗ python
Python 3.8.3 (default, Oct 14 2020, 10:33:18) 
[Clang 12.0.0 (clang-1200.0.32.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import onnxruntime
/Users/thomas/Documents/projects/pytorch-lightning/.venv/lib/python3.8/site-packages/onnxruntime/capi/_pybind_state.py:14: UserWarning: Cannot load onnxruntime.capi. Error: 'dlopen(/Users/thomas/Documents/projects/pytorch-lightning/.venv/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so, 2): Library not loaded: /usr/local/opt/libomp/lib/libomp.dylib
  Referenced from: /Users/thomas/Documents/projects/pytorch-lightning/.venv/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so
  Reason: image not found'.
  warnings.warn("Cannot load onnxruntime.capi. Error: '{0}'.".format(str(e)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/thomas/Documents/projects/pytorch-lightning/.venv/lib/python3.8/site-packages/onnxruntime/__init__.py", line 13, in <module>
    from onnxruntime.capi._pybind_state import get_all_providers, get_available_providers, get_device, set_seed, \
ImportError: cannot import name 'get_all_providers' from 'onnxruntime.capi._pybind_state' (/Users/thomas/Documents/projects/pytorch-lightning/.venv/lib/python3.8/site-packages/onnxruntime/capi/_pybind_state.py)
```

Comment (tchaton) [reply tests/models/test_onnx.py:21]: Hey @Borda, here is the bug I have locally. It is preventing me to `Discover Tests`.

Comment (Borda) [reply tests/models/test_onnx.py:21]: but it is happening only for you locally, or have you seen this also on CI?

**Pull request closed without merging**